### PR TITLE
UserAgentHeaderProvider unit test case added

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**

### CODE REVIEW

File: src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class UserAgentHeaderProviderTest {
+  @Test
+  public void getHeadersTest() {
+    UserAgentHeaderProvider userAgentHeaderProvider =
+        new UserAgentHeaderProvider("test agent header");
+    Map<String, String> resultMap = userAgentHeaderProvider.getHeaders();
+    String userAgent = resultMap.get("user-agent");
+    assertThat(resultMap).isNotNull();
+    assertThat(userAgent).isNotNull();
+    assertThat(resultMap instanceof Map).isTrue();
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java:33]: redundant test

File: src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    assertThat(userAgent instanceof String).isTrue();
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java:34]: redundant test

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
what about testing the `UserAgentProvider` ?

File: src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java (hunk 1 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java]: Removed

File: src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java (hunk 2 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/UserAgentHeaderProviderTest.java]: removed

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**